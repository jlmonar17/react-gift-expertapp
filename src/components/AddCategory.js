import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState("");

    const handleChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.length > 2) {
            setCategories((category) => [inputValue, ...category]);
            setInputValue("");
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" onChange={handleChange} value={inputValue} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};
