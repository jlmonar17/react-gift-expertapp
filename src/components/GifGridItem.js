export const GifGridItem = ({ id, title, url }) => {
    return (
        <div className="card">
            <img src={url} alt={title} key={id} />
            <p>{title}</p>
        </div>
    );
};
