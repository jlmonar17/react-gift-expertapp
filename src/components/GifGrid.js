import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__pulse">{category}</h3>

            {loading && (
                <p className="animate__animated animate__flash">Cargando...</p>
            )}

            <div className="card-grid">
                {images.map((image) => {
                    return <GifGridItem key={image.id} {...image} />;
                })}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
