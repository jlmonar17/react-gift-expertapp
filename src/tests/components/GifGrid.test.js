import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas para <GifGrid />", function () {
    const category = "Dragon Ball";

    test("Probando que GifGrid se renderice sin items cuando aun no se llama a las imágenes con useFetchGifs", () => {
        // Simulamos el retorno que tendrá esta función dentro de GifGrid, en este caso, con una data de gifs vacia
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("Debe de mostrar items cuando se cargan imagenes con useFetchGifs", () => {
        const gifs = [
            {
                id: "ABC",
                title: "Dragon Ball GIF",
                url: "https://localhost/imagen.jpg",
            },
            {
                id: "DEF",
                title: "Naruto GIF",
                url: "https://localhost/naruto.jpg",
            },
        ];

        // Simulamos el retorno que tendrá esta función dentro de GifGrid, en este caso, con una data de gifs
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        // Parrafo que dice "Cargando..." ya no debería aparacer
        expect(wrapper.find("p").exists()).toBe(false);
        // Los GifGridItem deben exister en al misma cantidad que el arreglo de gifs
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
    });
});
