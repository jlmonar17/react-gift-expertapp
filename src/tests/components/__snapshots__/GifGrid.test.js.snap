// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifGrid /> Debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Dragon Ball GIF"
      url="https://localhost/imagen.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Naruto GIF"
      url="https://localhost/naruto.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas para <GifGrid /> Probando que GifGrid se renderice sin items cuando aun no se llama a las im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
